File: Watchdog/src/main/kotlin/systemValues/RamUsage.kt
package systemValues

import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.Serializable
import kotlinx.serialization.protobuf.ProtoNumber

@ExperimentalSerializationApi
@Serializable
data class RamUsage(
    @ProtoNumber(1) val usedMb: Int = 0,
    @ProtoNumber(2) val totalMb: Int = 0
) : SystemValue

File: Agent/Agent.SystemValue.Api/Types/RamUsage.cs
using Agent.SystemValue.Api.Types.Base;
using ProtoBuf;

namespace Agent.SystemValue.Api.Types
{
    [ProtoContract]
    public class RamUsage : ISystemValue
    {
        [ProtoMember(1)] public uint UsedMb { get; }
        [ProtoMember(2)] public uint TotalMb { get; }

        public string Type => "metrics.ram.usage";

        public RamUsage()
        {
        }

        public RamUsage(uint usedMb, uint totalMb)
        {
            UsedMb = usedMb;
            TotalMb = totalMb;
        }

        public override string ToString() => $"RamUsage({UsedMb}/{TotalMb})";
    }
}